cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -2571,76 +2592,78 @@ public void trackEvent(String eventName, Float amount) {
-              if (properties != null) {
-                for (Map.Entry<String, Object> entry : properties.entrySet()) {
-                  propertiesObject.put(entry.getKey(), entry.getValue());
+              try {
+                JSONObject propertiesObject = new JSONObject();
+                if (properties != null) {
+                  for (Map.Entry<String, Object> entry : properties.entrySet()) {
+                    propertiesObject.put(entry.getKey(), entry.getValue());
+                  }
                 }
-              }
 
-              jsonBody.put("channelId", this.channelId);
-              jsonBody.put("eventId", eventId);
-              jsonBody.put("properties", propertiesObject);
-              jsonBody.put("subscriptionId", subscriptionId);
+                jsonBody.put("channelId", this.channelId);
+                jsonBody.put("eventId", eventId);
+                jsonBody.put("properties", propertiesObject);
+                jsonBody.put("subscriptionId", subscriptionId);
 
-              if (this.lastClickedNotificationId != null) {
-                jsonBody.put("notificationId", this.lastClickedNotificationId);
+                if (this.lastClickedNotificationId != null) {
+                  jsonBody.put("notificationId", this.lastClickedNotificationId);
+                }
+              } catch (JSONException ex) {
+                Logger.e(LOG_TAG, ex.getMessage(), ex);
               }
-            } catch (JSONException ex) {
-              Logger.e(LOG_TAG, ex.getMessage(), ex);
+
+              CleverPushHttpClient.postWithRetry("/subscription/conversion", jsonBody,
+                      new TrackEventResponseHandler().getResponseHandler(eventName));
+            } else {
+              Logger.d(LOG_TAG, "trackEvent: There is no subscription for CleverPush SDK.");
             }
+          });
 
-            CleverPushHttpClient.postWithRetry("/subscription/conversion", jsonBody,
-                new TrackEventResponseHandler().getResponseHandler(eventName));
-          } else {
-            Logger.d(LOG_TAG, "trackEvent: There is no subscription for CleverPush SDK.");
+          TriggeredEvent triggeredEvent = new TriggeredEvent(eventId, properties);
+          if (getAppBannerModule() == null) {
+            pendingAppBannerEvents.add(triggeredEvent);
+            return;
           }
-        });
-
-        TriggeredEvent triggeredEvent = new TriggeredEvent(eventId, properties);
-        if (getAppBannerModule() == null) {
-          pendingAppBannerEvents.add(triggeredEvent);
-          return;
-        }
-        getAppBannerModule().triggerEvent(triggeredEvent);
-      } catch (Exception ex) {
-        String message = "Track event failed because of error";
-        if (ex != null) {
-          message = ex.getMessage();
-          if (message == null) {
-            message = ex.toString();
+          getAppBannerModule().triggerEvent(triggeredEvent);
```

Comment:
This part should not wait for tracking consent. App Banner should work without tracking consent locally